### PR TITLE
Add SPI support to wiringX and implement for Hummingboard

### PR DESCRIPTION
**Pull request by crazystick**
With the dtsi file from
https://github.com/SolidRun/linux-imx6-3.14/pull/19 the SPI interface on the Hummingboard works.

Tested on Arch Linux Arm.

There is example code at https://gist.github.com/crazystick/4bd75a978fba91cfbea9

### CODE REVIEW

**Comment by CurlyMoo** [comment]
I'm happy to merge this PR, but it is possible to port the wiringPi SPI interace as well? Then we can be sure we found a cross-platform implementation.

**Comment by crazystick** [comment]
OK will do. I have a Pi somewhere around. I'll dig it out and give it a whirl.

**Comment by crazystick** [comment]
Now the gist posted earlier works on Hummingboard and RPi Revision 0003. From searching the forums, I can't see anyone saying they see SPI devices anywhere apart from /dev/spidev0.0 and /dev/spidev0.1, so I hope there is nothing to switch between RPi revisions.

**Comment by CurlyMoo** [comment]
On what devices did you test this?

**Comment by crazystick** [comment]
Tested on Hummingboard i2eXw and Raspberry Pi Revision 0003. I don’t have any other devices.

**Comment by CurlyMoo** [comment]
Is that a Raspberry Pi A, B, B+ or 2?

**Comment by crazystick** [comment]
According to the forums, its Model B Revision 1.0  Fuses mod and D14 removed

**Comment by CurlyMoo** [comment]
Is there a simple way i can test on the Raspberry Pi 2 and BananaPi?

**Comment by crazystick** [comment]
I used one of these: https://www.coolcomponents.co.uk/7-segment-serial-display-red.html and one of these: https://www.coolcomponents.co.uk/digital-temperature-sensor-breakout-tmp102.html. Its probably about the cheapest SPI and I2C device going.

**Comment by CurlyMoo** [comment]
Just one small comment. Can you adapt your coding style to mine. That means:

`if (...) {` > `if(...) {`
`spiSpeeds [2] ;` >  `spiSpeeds[2];`
`(unsigned long) data` > `(unsigned long)data`

File: raspberrypi.c [hunk 1]
```diff
@@ -820,6 +828,82 @@ static int raspberrypiI2CSetup(int devId) {
...
 	return fd;
 }
+
+int raspberrypiSPIGetFd(int channel) {
+	return spiFds[channel & 1];
+}
+
+int raspberrypiSPIDataRW(int channel, unsigned char *data, int len) {
+	struct spi_ioc_transfer spi;
+	memset (&spi, 0, sizeof(spi)); // found at http://www.raspberrypi.org/forums/viewtopic.php?p=680665#p680665
```
Comment (CurlyMoo) [inline raspberrypi.c]: Here is one left in regard to coding style...

**Comment by CurlyMoo** [comment]
Last step two steps:
1. update the gh-pages branch with the updated api.
2. update the python code with the new api.

I think both are pretty self-explanatory.

**Pull request merged**